src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,78 @@
-      <div className="modal-background" />
+  const [isModalLoading, setIsModalLoading] = useState(false);
+  const [user, setUser] = useState<User>();
 
-      <Loader />
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setIsModalLoading(true);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+    if (!currentTodo) {
+      return;
+    }
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+    getUser(currentTodo.userId)
+      .then(data => setUser({ ...data }))
+      .finally(() => setIsModalLoading(false));
+  }, [currentTodo]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  const handleCloseModal = () => {
+    dispatch(setCurrentTodo(null));
+  };
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+  return (
+    <div
+      className={classNames('modal', { 'is-active': currentTodo?.id !== 0 })}
+      data-cy="modal"
+    >
+      <div className="modal-background" />
+      {isModalLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{currentTodo?.id}
+            </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              onClick={handleCloseModal}
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+            />
+          </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
+            <p className="block" data-cy="modal-user">
+              {/* <strong className="has-text-success">Done</strong> */}
+              {currentTodo?.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
+              {' by '}
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
Ensure that 'user?.email' and 'user?.name' are defined before using them in the anchor tag to avoid rendering issues if 'user' is undefined.